client/src/main.rs

```diff
@@ -15,7 +15,7 @@ use structopt::StructOpt;
 struct Args {
     /// Admission Control port to connect to.
     #[structopt(short = "p", long = "port", default_value = "8000")]
-    pub port: String,
+    pub port: u16,
```

Comment:
Since we don't want the user to be able to supply a port number of `0` how about we change this type to be of type [`NonZeroU16`](https://doc.rust-lang.org/std/num/struct.NonZeroU16.html). That way you don't need to explicitly check for it like you do below.

Reply:
Thanks for your suggestion, the code is updated.